packages/smooth_app/lib/pages/product/ocr_widget.dart

```diff
@@ -0,0 +1,135 @@
+class OcrWidget extends StatelessWidget {
+  const OcrWidget({
+    required this.controller,
+    required this.onSubmitField,
+    required this.onTapNewImage,
+    required this.onTapExtractData,
+    required this.productImageData,
+    required this.product,
+    required this.helper,
+  });
+
+  final TextEditingController controller;
+  final Future<void> Function() onTapNewImage;
+  final Future<void> Function() onTapExtractData;
+  final Future<void> Function(ImageField) onSubmitField;
+  final ProductImageData productImageData;
+  final Product product;
+  final OcrHelper helper;
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    return Align(
+      alignment: AlignmentDirectional.bottomStart,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: <Widget>[
+          Flexible(
+            flex: 1,
+            child: Align(
+              alignment: Alignment.bottomRight,
+              child: Padding(
+                padding: const EdgeInsetsDirectional.only(
+                  bottom: LARGE_SPACE,
+                  start: LARGE_SPACE,
+                  end: LARGE_SPACE,
+                ),
+                child: SmoothActionButtonsBar(
+                  positiveAction: SmoothActionButton(
+                    text: (TransientFile.isImageAvailable(
+                      productImageData,
+                      product.barcode!,
+                    ))
+                        ? helper.getActionRefreshPhoto(appLocalizations)
+                        : appLocalizations.upload_image,
+                    onPressed: () async => onTapNewImage(),
+                  ),
+                ),
+              ),
+            ),
+          ),
+          Flexible(
+            flex: 1,
+            child: SingleChildScrollView(
+              child: Container(
+                decoration: BoxDecoration(
+                    color: Theme.of(context).colorScheme.background,
+                    borderRadius: const BorderRadius.only(
+                      topLeft: ANGULAR_RADIUS,
+                      topRight: ANGULAR_RADIUS,
+                    )),
+                child: Padding(
+                  padding: const EdgeInsets.all(LARGE_SPACE),
+                  child: Column(
+                    children: <Widget>[
+                      if (TransientFile.isServerImage(
+                        productImageData,
+                        product.barcode!,
+                      ))
+                        SmoothActionButtonsBar.single(
+                          action: SmoothActionButton(
+                            text: helper.getActionExtractText(appLocalizations),
+                            onPressed: () async => onTapExtractData(),
+                          ),
+                        )
+                      else
+                        const CircularProgressIndicator.adaptive(),
```

Comment:
ImageUploads can take some time, showing a "Image is still uploding" would make it more user friendly. A todo in this PR would be fine